tests/integration/nvidia_tuned/test_prepare_nvidia_profiles.py

```
@@ -877,3 +879,474 @@ def test_prepare_nvidia_profiles_vr200_bcm_no_bootloader(base_image, intent):
+    """configure_bootloader.sh writes a drop-in that resolves to the profile cmdline."""
+    runner = DockerTestRunner(package="nvidia-tuned", base_image=base_image)
+    try:
+        configmaps = {"accelerator": "gb200", "intent": "performance", "service": "rke2"}
+        create_container_for_testing(runner, configmaps)
+        _install_grub_stub(runner)
+        _write_bootcmdline(runner, STUB_BOOTCMDLINE, SAMPLE_CMDLINE)
+
+        result = _run_with_env(runner, "configure_bootloader.sh", STUB_ENV)
+        assert_exit_code(result, 0)
+        assert runner.file_exists(STUB_DROPIN), f"drop-in was not written to {STUB_DROPIN}"
+        assert runner.file_exists("/tmp/update-grub.ran"), "grub was not regenerated"
```

Ruff (0.16.3)

[error] Probable insecure usage of temporary file or directory: "/tmp/update-grub.ran"

(S108)